tests/test_postgres_incremental_replication.py

```diff
@@ -249,369 +343,449 @@ def get_properties():
-                             'our_integer': 44101,
-                             'our_double': decimal.Decimal('1.1'),
-                             'id': 2,
-                             'our_json': '{"nymn": 77}',
-                             'our_boolean': True,
-                             'our_jsonb': '{"burgers": "good++"}',
-                             'our_bigint': 1000001,
-                             'OUR TIME TZ': '10:09:08-04:00',
-                             'our_store': {"name" : "betty", "dances" :"floor"},
-                             'OUR TS TZ': '1987-03-03T08:03:03.733184+00:00',
-                             'our_smallint': 2,
-                             'OUR DATE':     '1964-07-01T00:00:00+00:00',
-                             'our_varchar':  'our_varchar 2',
-                             'OUR TS':       '1987-03-03T03:03:03.733184+00:00',
-                             'our_uuid':     self.rec_2['our_uuid'],
-                             'our_real':     decimal.Decimal('1.2'),
-                             'our_varchar_10': 'varchar_10',
-                             'our_citext'  : self.rec_2['our_citext'],
-                             'our_inet'    : self.rec_2['our_inet'],
-                             'our_cidr'    : self.rec_2['our_cidr'],
-                             'our_mac'     : self.rec_2['our_mac'],
-                             'our_money'      : None}
-
-
-
-        actual_record_1 = records_by_stream[test_table_name]['messages'][1]
-        self.assertEqual(set(actual_record_1['data'].keys()), set(expected_record_1.keys()),
-                         msg="keys for expected_record_1 are wrong: {}".format(set(actual_record_1.keys()).symmetric_difference(set(expected_record_1.keys()))))
-
-        for k in actual_record_1['data'].keys():
-            self.assertEqual(actual_record_1['data'][k], expected_record_1[k], msg="{} != {} for key {}".format(actual_record_1['data'][k], expected_record_1[k], k))
-
-        actual_record_2 = records_by_stream[test_table_name]['messages'][2]
-        self.assertEqual(set(actual_record_1['data'].keys()), set(expected_record_2.keys()),
-                         msg="keys for expected_record_2 are wrong: {}".format(set(actual_record_2.keys()).symmetric_difference(set(expected_record_2.keys()))))
-
-        for k in actual_record_2['data'].keys():
-            self.assertEqual(actual_record_2['data'][k], expected_record_2[k], msg="{} != {} for key {}".format(actual_record_2['data'][k], expected_record_2[k], k))
+        # verify the persisted schema matches expectations
+        self.assertEqual(expected_schemas[test_table_name], records_by_stream[test_table_name]['schema'])
+
+        # verify replicated records match expectations
+        self.assertDictEqual(self.expected_records[0], messages[1]['data'])
+        self.assertDictEqual(self.expected_records[1], messages[2]['data'])
+        self.assertDictEqual(self.expected_records[2], messages[3]['data'])
```

Comment:
Is order of the messages part of the expectations?  Like we are testing that they are sent in order of the replication key?  I think that is a valid expectation but I'm not 100%.

Reply:
Yes, they are in order of replication key. I can add an assertion that each rep key is greater than the one before it to give greater confidence.

Reply:
TODO added to branch `testing/replication-pt2`.